### PR TITLE
Chibitronics v0.1.40 release blog

### PR DESCRIPTION
**Pull request by microsoftsam**
Chibitronics v0.1.40 release blog post. 

Covered topics: 
- [x] New blocks (APIs)
- [x] Deprecated blocks
- [x] Scratch blocks change
- [x] Field editors
- [x] New packages
- [x] Home screen

### CODE REVIEW

File: docs/blog.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Microsoft MakeCode Blog
 
+## [Chibitronics Editor v0.1.40 (Scratch blocks)](/blog/chibitronics/v0.1.40)
+
+October 8th, 2017
```
Comment (pelikhan) [inline docs/blog.md]: Add your name

File: docs/blog/chibitronics/v0.1.40.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+All inputs either have a ROUND or HEXAGONAL shape. 
+If a block has a return type, it will have a shape: 
+- Boolean: HEXAGONAL
+- Number, String, Other: ROUND
+
+![Scratch blocks different shaped blocks](/static/blog/chibitronics/v0.1.40/scratchblocksshapes.png)
+
+
+We've based our Blockly fork (pxt-blockly) against [scratch-blocks](https://github.com/LLK/scratch-blocks).
```
Comment (pelikhan) [inline docs/blog/chibitronics/v0.1.40.md]: our **open source** ...

File: docs/blog/chibitronics/v0.1.40.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+- SOS
+
+Be sure to check them out!
+
+![show effect block screenshot](/static/blog/chibitronics/v0.1.40/showeffectblock.png)
+
+
+## Deprecated blocks
+
+We have deprecated the **set ON** and **set OFF** blocks in favour of the new **turn ON/OFF** block.
```
Comment (pelikhan) [inline docs/blog/chibitronics/v0.1.40.md]: They will still work but you won’t be able to create new ones.

**Comment by tballmsft** [comment]
Actually, change

Scratch-blocks

to 

Scratch Blocks

File: docs/blog.md [hunk 4]
```diff
@@ -1,5 +1,12 @@
...
+
+A big release for our Chibitronics editor!
+**[Read more](/blog/chibitronics/v0.1.40)**
```
Comment (pelikhan) [inline docs/blog.md]: Should be 0.2 since you had API changes?

File: docs/blog/chibitronics/v0.1.40.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+**Posted on October 8, 2017 by [microsoftsam](https://github.com/microsoftsam)**
+
+We are very excited about all the new features we're unveiling this release. 
+This update brings **Scratch** style rendering of the blocks, a suite of new field editors to enhance the editing experience, a new home screen, a number of simulator fixes and much much more.
+
+
+## Scratch blocks
+
+In this release, we have re-vamped the look of our blocks and aligned with **Scratch**.
```
Comment (pelikhan) [inline docs/blog/chibitronics/v0.1.40.md]: Scratch Blocks

File: docs/blog/chibitronics/v0.1.40.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+The number slider also got a face-lift, and now it applies the parent block's color to the slider.
+
+![Number slider screenshot](/static/blog/chibitronics/v0.1.40/field_slider.png)
+
+
+## New Packages
+
+The Chibi SCOPE blocks have moved into their own package. In projects where you'd like to use the Chibi SCOPE, you will now have to go to "Advanced" in the toolbox and click "Add Package". This will bring up the "Add Package" dialog, select or search for the "Chibi SCOPE" package and click on it to add it to your project.
+
+Packages allow us to create many independent extensions to the editor that include their own set of blocks. Since the Chibi SCOPE is a separate piece of hardware, we decided to encapsulate it's blocks into a package.
```
Comment (pelikhan) [inline docs/blog/chibitronics/v0.1.40.md]: Add image + link to Chibi scope

File: docs/blog.md (hunk 4 above)
Comment (microsoftsam) [reply docs/blog.md]: Last released version was 0.0.60 so were good

File: docs/blog/chibitronics/v0.1.40.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
+# Chibitronics Editor v0.1.40
+
+**Posted on October 8, 2017 by [microsoftsam](https://github.com/microsoftsam)**
+
+We are very excited about all the new features we're unveiling this release. 
+This update brings **Scratch** style rendering of the blocks, a suite of new field editors to enhance the editing experience, a new home screen, a number of simulator fixes and much much more.
```
Comment (ganicke) [inline docs/blog/chibitronics/v0.1.40.md]: one 'much' is good enough.

File: docs/blog/chibitronics/v0.1.40.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+
+## Scratch blocks
+
+In this release, we have re-vamped the look of our blocks and aligned with **Scratch**. 
+
+Besides the change in the aesthetic look of the blocks, the main difference here is that the Blockly input (reporter) puzzle piece no longer exists.
```
Comment (ganicke) [inline docs/blog/chibitronics/v0.1.40.md]: 'Besides the changes to' or Besides a change to'

File: docs/blog/chibitronics/v0.1.40.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+![Scratch blocks different shaped blocks](/static/blog/chibitronics/v0.1.40/scratchblocksshapes.png)
+
+
+We've based our open source Blockly fork (pxt-blockly) against [scratch-blocks](https://github.com/LLK/scratch-blocks). 
+
+## New Blocks
+
+We have introduced two new APIs to make the editing experience even simpler. 
+
+**turn PIN ON** block:
```
Comment (ganicke) [inline docs/blog/chibitronics/v0.1.40.md]: Just say **turn PIN ON:**. Block is implied and mentioned in the description following.

File: docs/blog/chibitronics/v0.1.40.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+**turn PIN ON** block: 
+The turn PIN ON/OFF block can be used to switch a pin ON or OFF. Under the hood, the API sets the pin output mode to INPUT / OUTPUT and calls digitalWrite with either 0 or 1.
+
+The Toggle field editor on the block makes it easy to switch between on and off inline.
+
+![turn ON/OFF block screenshot](/static/blog/chibitronics/v0.1.40/turnonoff.png)
+
+
+**show effect** block: 
+The show effect block has a number of pre-programmed effects at it's disposal. You can select the effect you want to use and the pin you want to apply it on, and the built-in effect will run on that PIN.
```
Comment (ganicke) [inline docs/blog/chibitronics/v0.1.40.md]: ...apply it on. The built-in effect will run...

File: docs/blog/chibitronics/v0.1.40.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+Over the past few months, we have invested heavily in our suite of field editors. Today we are happy to include 5 new field editors to the list that can be used by targets.
+
+### String fields: 
+
+All inputs of type string use the string field editor, to allow us to distinguish between a String and a Number. A string field editor shows quotes around the input.
+
+![Field string screenshot](/static/blog/chibitronics/v0.1.40/field_string.png)
+
+### Number dropdown:
+The number dropdown field allows you to edit a number value but also gives the user a list of options to hint to them the sort of values they could expect to use.
```
Comment (ganicke) [inline docs/blog/chibitronics/v0.1.40.md]: You mean this?:

The number dropdown field allows you to edit a number value but also gives you a list of options which are hints for the types of values you might expect to use.

File: docs/blog/chibitronics/v0.1.40.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+![Color wheel screenshot](/static/blog/chibitronics/v0.1.40/field_colorwheel.png)
+
+### Color picker:
+The Color picker allows you to choose from a list of pre-defined colors. We have chosen some great colors that work well for the Neopixel light onboard the Chibi Chip.
+
+![Color picker screenshot](/static/blog/chibitronics/v0.1.40/field_colorpicker.png)
+
+### Number slider:
+The number slider also got a face-lift, and now it applies the parent block's color to the slider.
```
Comment (ganicke) [inline docs/blog/chibitronics/v0.1.40.md]: The number slider also got a face-lift. It now applies the parent block's color to the slider.

File: docs/blog/chibitronics/v0.1.40.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+## New Packages
+
+The Chibi SCOPE blocks have moved into their own package. In projects where you'd like to use the Chibi SCOPE, you will now have to go to "Advanced" in the toolbox and click "Add Package". This will bring up the "Add Package" dialog, select or search for the "Chibi SCOPE" package and click on it to add it to your project.
+
+Packages allow us to create many independent extensions to the editor that include their own set of blocks. Since the Chibi SCOPE is a separate piece of hardware, we decided to encapsulate it's blocks into a package. 
+
+## Home screen
+
+We have introduced another new feature in this release, the home screen. This screen acts as the primary navigation point, you can use it to switch between projects, create new projects, start tutorials, or browser for ideas.
```
Comment (ganicke) [inline docs/blog/chibitronics/v0.1.40.md]: ..., or browse for ideas.

**Comment by Jaqster** [comment]
I would add a paragraph up-front setting context - what is Chibitronics, why is it a cool and innovative partner?  Also, don't say "aligned with Scratch" - that's not true, we're just leveraging some of the UI components and rendering improvements that come with the new version of Scratch Blocks 3.0.  Also, let's not say we've based our fork against Scratch Blocks - doesn't give us enough credit.  We should also talk about the user testing/workshops we've done that have informed these changes (we didn't just make this up).  I'll take a pass at some edits tomorrow.

**Comment by microsoftsam** [comment]
@Jaqster that sounds good! Sounds like you have a good idea of what should go in that paragraph, feel free to make that edits directly :)

**Pull request merged**